packages/api/src/platforms/vtex/resolvers/query.ts

```
@@ -393,13 +393,18 @@ export const Query = {
 
     try {
       const order = await commerce.oms.userOrder({ orderId })
-      const commercialAuth =
-        await commerce.oms.getCommercialAuthorizationsByOrderId({ orderId })
 
       if (!order) {
         throw new NotFoundError(`No order found for id ${orderId}`)
       }
 
+      const commercialAuth =
+        await commerce.oms.getCommercialAuthorizationsByOrderId({ orderId })
+
+      if (!commercialAuth) {
+        throw new NotFoundError(`OrderAuth for ${orderId} not found`)
+      }
+
```

I don't believe this should throw an error.
The order page won't render because of this, but in the case where it doesn't have any rule to approve it means the approve flow isn't necessary. So, the `userOrder` should return just fine.